From 31afda6b44c432025587b7152db1bb63a4137c2d Mon Sep 17 00:00:00 2001
From: Sebastian Schmidt <yath@yath.de>
Date: Fri, 8 Jun 2012 19:33:03 +0200
Subject: [PATCH 2/4] Allow passing of perl libraries with $PERL_LIBS

---
 configure.in    |   13 +++++++++++++
 src/Makefile.am |    1 +
 2 files changed, 14 insertions(+)

diff --git a/configure.in b/configure.in
index dfd980e..a5fb686 100644
--- a/configure.in
+++ b/configure.in
@@ -2944,6 +2944,7 @@ if test "x$with_libperl" = "xyes" \
 then
   SAVE_CFLAGS="$CFLAGS"
   SAVE_LDFLAGS="$LDFLAGS"
+  SAVE_LIBS="$LIBS"
 dnl ARCHFLAGS="" -> disable multi -arch on OSX (see Config_heavy.pl:fetch_string)
   if test "x$PERL_CFLAGS" = "x"
   then
@@ -2955,6 +2956,7 @@ dnl ARCHFLAGS="" -> disable multi -arch on OSX (see Config_heavy.pl:fetch_string
   fi
   CFLAGS="$CFLAGS $PERL_CFLAGS"
   LDFLAGS="$LDFLAGS $PERL_LDFLAGS"
+  LIBS="$LIBS $PERL_LIBS"
 
   AC_CACHE_CHECK([for libperl],
     [c_cv_have_libperl],
@@ -2982,12 +2984,14 @@ dnl ARCHFLAGS="" -> disable multi -arch on OSX (see Config_heavy.pl:fetch_string
 	  AC_DEFINE(HAVE_LIBPERL, 1, [Define if libperl is present and usable.])
 	  AC_SUBST(PERL_CFLAGS)
 	  AC_SUBST(PERL_LDFLAGS)
+      AC_SUBST(PERL_LIBS)
   else
 	  with_libperl="no"
   fi
 
   CFLAGS="$SAVE_CFLAGS"
   LDFLAGS="$SAVE_LDFLAGS"
+  LIBS="$SAVE_LIBS"
 else if test -z "$perl_interpreter"; then
   with_libperl="no (no perl interpreter found)"
   c_cv_have_libperl="no"
@@ -2998,8 +3002,10 @@ if test "x$with_libperl" = "xyes"
 then
 	SAVE_CFLAGS="$CFLAGS"
 	SAVE_LDFLAGS="$LDFLAGS"
+	SAVE_LIBS="$LIBS"
 	CFLAGS="$CFLAGS $PERL_CFLAGS"
 	LDFLAGS="$LDFLAGS $PERL_LDFLAGS"
+	LIBS="$LIBS $PERL_LIBS"
 
 	AC_CACHE_CHECK([if perl supports ithreads],
 		[c_cv_have_perl_ithreads],
@@ -3027,16 +3033,19 @@ then
 
 	CFLAGS="$SAVE_CFLAGS"
 	LDFLAGS="$SAVE_LDFLAGS"
+	LIBS="$SAVE_LIBS"
 fi
 
 if test "x$with_libperl" = "xyes"
 then
 	SAVE_CFLAGS="$CFLAGS"
 	SAVE_LDFLAGS="$LDFLAGS"
+	SAVE_LIBS="$LIBS"
 	# trigger an error if Perl_load_module*() uses __attribute__nonnull__(3)
 	# (see issues #41 and #42)
 	CFLAGS="$CFLAGS $PERL_CFLAGS -Wall -Werror"
 	LDFLAGS="$LDFLAGS $PERL_LDFLAGS"
+	LIBS="$LIBS $PERL_LIBS"
 
 	AC_CACHE_CHECK([for broken Perl_load_module()],
 		[c_cv_have_broken_perl_load_module],
@@ -3061,6 +3070,7 @@ then
 
 	CFLAGS="$SAVE_CFLAGS"
 	LDFLAGS="$SAVE_LDFLAGS"
+	LIBS="$SAVE_LIBS"
 fi
 AM_CONDITIONAL(HAVE_BROKEN_PERL_LOAD_MODULE,
 		test "x$c_cv_have_broken_perl_load_module" = "xyes")
@@ -3069,8 +3079,10 @@ if test "x$with_libperl" = "xyes"
 then
 	SAVE_CFLAGS="$CFLAGS"
 	SAVE_LDFLAGS="$LDFLAGS"
+	SAVE_LIBS="$LIBS"
 	CFLAGS="$CFLAGS $PERL_CFLAGS"
 	LDFLAGS="$LDFLAGS $PERL_LDFLAGS"
+	LIBS="$LIBS $PERL_LIBS"
 
 	AC_CHECK_MEMBER(
 		[struct mgvtbl.svt_local],
@@ -3090,6 +3102,7 @@ then
 
 	CFLAGS="$SAVE_CFLAGS"
 	LDFLAGS="$SAVE_LDFLAGS"
+	LIBS="$SAVE_LIBS"
 fi
 # }}}
 
diff --git a/src/Makefile.am b/src/Makefile.am
index f106fa1..1b912aa 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -858,6 +858,7 @@ perl_la_CFLAGS += -Wno-nonnull
 endif
 perl_la_LDFLAGS = -module -avoid-version \
 		$(PERL_LDFLAGS)
+perl_la_LIBADD = $(PERL_LIBS)
 collectd_LDADD += "-dlopen" perl.la
 collectd_DEPENDENCIES += perl.la
 endif
-- 
1.7.10

